src/mam/mod_mam.erl

```diff
@@ -160,64 +161,57 @@
+    Entries = mongoose_hooks:get_mam_pm_gdpr_data(HostType, ArcJID),
     [{mam_pm, Schema, Entries} | Acc].
 
 -spec delete_archive(jid:server(), jid:user()) -> 'ok'.
 delete_archive(Server, User)
   when is_binary(Server), is_binary(User) ->
     ?LOG_DEBUG(#{what => mam_delete_archive, user => User, server => Server}),
     ArcJID = jid:make(User, Server, <<>>),
-    Host = server_host(ArcJID),
-    ArcID = archive_id_int(Host, ArcJID),
-    remove_archive_hook(Host, ArcID, ArcJID),
+    HostType = jid_to_host_type(ArcJID),
+    ArcID = archive_id_int(HostType, ArcJID),
+    remove_archive_hook(HostType, ArcID, ArcJID),
     ok.
 
-
 -spec archive_size(jid:server(), jid:user()) -> integer().
 archive_size(Server, User)
   when is_binary(Server), is_binary(User) ->
     ArcJID = jid:make(User, Server, <<>>),
-    Host = server_host(ArcJID),
-    ArcID = archive_id_int(Host, ArcJID),
-    archive_size(Host, ArcID, ArcJID).
-
+    HostType = jid_to_host_type(ArcJID),
+    ArcID = archive_id_int(HostType, ArcJID),
+    archive_size(HostType, ArcID, ArcJID).
 
 -spec archive_id(jid:server(), jid:user()) -> integer() | undefined.
 archive_id(Server, User)
   when is_binary(Server), is_binary(User) ->
     ArcJID = jid:make(User, Server, <<>>),
-    Host = server_host(ArcJID),
-    archive_id_int(Host, ArcJID).
+    HostType = jid_to_host_type(ArcJID),
+    archive_id_int(HostType, ArcJID).
 
 %% gen_mod callbacks
 %% Starting and stopping functions for users' archives
 
--spec start(Host :: jid:server(), Opts :: list()) -> any().
-start(Host, Opts) ->
+-spec start(HostType :: host_type(), Opts :: list()) -> any().
+start(HostType, Opts) ->
     ?LOG_INFO(#{what => mam_starting}),
-
-    %% `parallel' is the only one recommended here.
-    IQDisc = gen_mod:get_opt(iqdisc, Opts, parallel), %% Type
-    [mod_disco:register_feature(Host, Feature) || Feature <- features(?MODULE, Host)],
-    gen_iq_handler:add_iq_handler(ejabberd_sm, Host, ?NS_MAM_04,
-                                  ?MODULE, process_mam_iq, IQDisc),
-    gen_iq_handler:add_iq_handler(ejabberd_sm, Host, ?NS_MAM_06,
-                                  ?MODULE, process_mam_iq, IQDisc),
-    ejabberd_hooks:add(hooks(Host)),
-    ensure_metrics(Host),
+    ?LOG_ERROR(#{what => mam_starting, host_type => HostType}),
+    ensure_metrics(HostType),
+    ejabberd_hooks:add(hooks(HostType)),
+    add_id_handlers(HostType, Opts),
+    register_features(HostType),
     ok.
 
-
--spec stop(Host :: jid:server()) -> any().
-stop(Host) ->
+-spec stop(HostType :: host_type()) -> any().
+stop(HostType) ->
     ?LOG_INFO(#{what => mam_stopping}),
-    ejabberd_hooks:delete(hooks(Host)),
-    gen_iq_handler:remove_iq_handler(ejabberd_sm, Host, ?NS_MAM_04),
-    gen_iq_handler:remove_iq_handler(ejabberd_sm, Host, ?NS_MAM_06),
-    [mod_disco:unregister_feature(Host, Feature) || Feature <- features(?MODULE, Host)],
+    ?LOG_ERROR(#{what => mam_stopping, host_type => HostType}),
```

Comment:
More leftovers. The host_type could actually be a field of the previous LOG_INFO. But surely this shouldn't be an error.